### ISSUE TITLE
Left 4 Dead 2 - fullscreen error with dxvk-native

### ISSUE THREAD

**Issue by vlad54rus** [doitsujin/dxvk#3083]
Please describe your issue as accurately as possible.

Left 4 Dead 2 fails to start in fullscreen mode when using dxvk-native, only works in borderless and windowed modes. This doesn't happen with regular dxvk.
If launched in windowed first then switched to fullscreen - rendering stops and just flickers between two last frames.

### Software information
Left 4 Dead 2, but i've seen people also having this issue in Portal 2 ([example1](https://steamcommunity.com/app/620/eventcomments/3548301990172367052?ctp=3#c3548301990172786484), [example2](https://steamcommunity.com/app/620/discussions/0/6204299339028748164/)).

### System information
- GPU: NVIDIA GTX 1070
- Driver: 516.89.0
- OS: Windows 10
- DXVK version: native-1.9.1a-4-g37690dde+

### Apitrace file(s)
- https://drive.google.com/file/d/1xz20jo50KaNDQa9Stx03P8y-hAQUNUU1/view?usp=sharing

### Log files
- d3d9.log:
[left4dead2_d3d9(native).log](https://github.com/doitsujin/dxvk/files/10046701/left4dead2_d3d9.native.log)
[left4dead2_d3d9(2.0).log](https://github.com/doitsujin/dxvk/files/10046702/left4dead2_d3d9.2.0.log)

**Comment by Blisto91** [comment]
@vlad54rus Left 4 Dead 2 should have received a update to dxvk 2.0 today (i think it's also on windows at least) is it still the same with that?

**Comment by vlad54rus** [comment]
It's Windows. Got the update after restarting Steam but no dxvk 2.0 with it - still the same 1-yeal old dxvk_d3d9.dll which the game uses when passing -vulkan launch option.

**Comment by Blisto91** [comment]
Ah yes i see. Dxvk was only updated on the Linux version https://steamdb.info/patchnotes/9969999/

**Comment by Joshua-Ashton** [comment]
This is an Windows exclusive NVIDIA driver bug that has yet to be fixed.

Are you on an optimus laptop @vlad54rus ?

**Comment by vlad54rus** [comment]
It's a desktop and NVIDIA is the only GPU.
What bug exactly and why doesn't this happen with regular dxvk?

**Comment by Blisto91** [comment]
I'm not sure I understand when you say regular dxvk.
Do you mean you download it here from GitHub and then put next to the exe or do you replace the game bundled file and launch with Vulkan?

**Comment by vlad54rus** [comment]
Yes, downloaded from here and replaced the bundled file.

**Comment by McMasterGold** [comment]
Getting the exact same issue with Portal 2, using a GTX 750 on desktop. Exclusive fullscreen on Vulkan worked just fine before the DXVK 2.0 patch.

Also now borderless windowed exhibits tearing, even with in-game V-Sync enabled.

**Comment by K0bin** [comment]
> Also now borderless windowed exhibits tearing, even with in-game V-Sync enabled.

That sounds like a driver issue.

**Comment by McMasterGold** [comment]
> That sounds like a driver issue.
I see. I definitely didn't have any tearing with borderless windowed before the DXVK 2.0 patch, but then I did update to the latest ForceWare 526.98 a few days ago. I'll try rolling back and see if it helps things.

**Comment by Blisto91** [comment]
@McMasterGold did you manually replace dxvk?

**Comment by McMasterGold** [comment]
> @McMasterGold did you manually replace dxvk?

No, I'm simply launching the game with the -vulkan parameter. If I'm reading the above comments correctly, replacing the DLLs with the ones from here fixes the issue? I'll give that a try.

**Comment by Blisto91** [comment]
Roger. From what I've gathered only the Linux version have been updated to 2.0 whereas Windows still has the old version. Which for some reason the changelog gives no indication of what so ever.
And yes you can replace the dll manually

**Comment by McMasterGold** [comment]
> Roger. From what I've gathered (seen on steamdb) only the Linux version have been updated to 2.0 whereas Windows still has the old version. Which for some reason the changelog gives no indication of what so ever. And yes you can replace the dll manually

So the Windows version didn't get changed at all? Makes the new issues that surfaced even more bizarre. Perhaps it is a driver problem after all.

I replaced the DLL with the newest one and thankfully exclusive fullscreen is working again now. Still getting tearing with borderless windowed, but I prefer fullscreen so not a problem for me. The new version seems to be doing an even better job at reducing the DX9 stutters as well. Thanks for the tips!

**Comment by vlad54rus** [comment]
Vsync in windowed doesn't work because ingame setting only applies it to fullscreen (could be the same for other Source games too). You can force vsync from dxvk.conf instead.

**Comment by Blisto91** [comment]
So since this seems to be working with a manual replacement of the bundled Windows dxvk to 2.0 I'm guessing it should get solved if Valve updates the Windows version @Joshua-Ashton

**Comment by McMasterGold** [comment]
> Vsync in windowed doesn't work because ingame setting only applies it to fullscreen (could be the same for other Source games too). You can force vsync from dxvk.conf instead.

Can confirm this works great on my end. Thanks for the useful tip!

**Comment by shoober420** [comment]
This also occurs with Portal 2 as well. When I replace the original dxvk_dx9.dll with the new x32/d3d9.dll and rename it to match original, exclusive fullscreen works. This does not appear to happen with Half-Life 2 using -vulkan launch option and DXVK.

Windows 11 27686
Intel i9 13900kf
NVIDIA RTX 4090
Vulkan beta driver: 553.00

**Comment by shoober420** [comment]
This also occurs with Portal 2 as well. When I replace the original dxvk_dx9.dll with the new x32/d3d9.dll and rename it to match original, exclusive fullscreen works. This does not appear to happen with Half-Life 2 using -vulkan launch option and DXVK.

Windows 11 27686
Intel i9 13900kf
NVIDIA RTX 4090
Vulkan beta driver: 553.00

**Comment by Blisto91** [comment]
@Joshua-Ashton Friendly ping.

**Comment by Blisto91** [comment]
Since this issue is solved with an up to date version of dxvk i am going to close it and refer to https://github.com/ValveSoftware/Source-1-Games for requests of updating the bundled version.
